src/main/java/io/confluent/idesidecar/restapi/proxy/ConnectionProcessor.java

```diff
@@ -12,9 +12,9 @@
  * Processor that checks for the `x-connection-id` header and retrieves the connection state from
  * the connection state manager.
  *
- * @param <T> The proxy context type, which must extend {@link ProxyContext}
+ * @param <T> The proxy context type, which must extend {@link io.confluent.idesidecar.restapi.proxy.ClusterProxyContext}
  */
-public class ConnectionProcessor<T extends ProxyContext> extends
+public class ConnectionProcessor<T extends ClusterProxyContext> extends
```

Comment:
The generic type `T` needed to have context about the cluster type in order to set TLS configs appropriately.

This triggered a refactor to essentially move up all the fields from `ClusterProxyContext` into `ProxyContext`, and then rename `ProxyContext` to `ClusterProxyContext`.

Reply:
I'm going to revert this change. Should substantially reduce the size of this PR's diff.